api/base/views.py

```diff
@@ -75,3 +79,21 @@ def root(request, format=None):
             'users': absolute_reverse('users:user-list'),
         }
     })
+
+
+class OsfAPIViewMeta(type):
+    """Metaclass to ensure a base level of permissions are applied to all of our views
+    """
+    def __new__(cls, name, bases, attributes):
+        permission_classes = set((
+            drf_permissions.IsAuthenticatedOrReadOnly,
+            base_permissions.TokenHasScope,
+        ))
+        bases_permission_classes = attributes.get('permission_classes', tuple())
+        try:
+            iter(bases_permission_classes)
+        except TypeError:
+            bases_permission_classes = (bases_permission_classes, )
+        bases_permission_classes = set(bases_permission_classes)
```

Comment:
I meant overriding `__init__` of the metaclass. `self` will be the *instance of the class object*; at that point, the class has been instantiated, so class attributes will have been inherited.